File: sdk/nodejs/src/resources/qr_and_thanh_toan.js
'use strict';

class QrAndThanhToanResource {
  constructor(http) { this._http = http; }

  /** GET /api/banks - Danh bạ ngân hàng Việt Nam (65 ngân hàng): mã... */
  banks(params) { return this._http.get('/api/banks', params); }

  /** GET /api/banks/2 - Tra một ngân hàng theo mã BIN (970436) */
  banks_2(params) { return this._http.get('/api/banks/2', params); }

  /** GET /api/generator/vietqr - Sinh mã VietQR theo chuẩn NAPAS để nhận chuyển... */
  generator_vietqr(params) { return this._http.get('/api/generator/vietqr', params); }

  /** POST /api/tools/qr/generate - Sinh mã QR từ nội dung bất kỳ */
  tools_qr_generate(data) { return this._http.post('/api/tools/qr/generate', data); }

  /** POST /api/tools/qr/vcard - Sinh mã QR danh thiếp vCard */
  tools_qr_vcard(data) { return this._http.post('/api/tools/qr/vcard', data); }

}

module.exports = { QrAndThanhToanResource };

File: sdk/go/hitechcloudtools/qr_and_thanh_toan.go
package hitechcloudtools

// QrAndThanhToanResource - QR & Thanh toán (5 endpoints)
type QrAndThanhToanResource struct {
	client *Client
}

// Banks - Danh bạ ngân hàng Việt Nam (65 ngân hàng): mã...
func (r *QrAndThanhToanResource) Banks(params map[string]string) (map[string]interface{}, error) {
	return r.client.Get("/api/banks", params)
}

// Banks2 - Tra một ngân hàng theo mã BIN (970436)
func (r *QrAndThanhToanResource) Banks2(params map[string]string) (map[string]interface{}, error) {
	return r.client.Get("/api/banks/2", params)
}

// GeneratorVietqr - Sinh mã VietQR theo chuẩn NAPAS để nhận chuyển...
func (r *QrAndThanhToanResource) GeneratorVietqr(params map[string]string) (map[string]interface{}, error) {
	return r.client.Get("/api/generator/vietqr", params)
}

// ToolsQrGenerate - Sinh mã QR từ nội dung bất kỳ
func (r *QrAndThanhToanResource) ToolsQrGenerate(body interface{}) (map[string]interface{}, error) {
	return r.client.Post("/api/tools/qr/generate", body)
}

// ToolsQrVcard - Sinh mã QR danh thiếp vCard
func (r *QrAndThanhToanResource) ToolsQrVcard(body interface{}) (map[string]interface{}, error) {
	return r.client.Post("/api/tools/qr/vcard", body)
}

